src/descriptor/mod.rs

```diff
@@ -1022,7 +1022,8 @@ impl_from_str!(
                 Ok(tr) => Ok(Descriptor::Tr(tr)),
                 Err(_) => {
                     // Try parsing with extensions
-                    let tr = Tr::<Pk, T>::from_str(s)?;
+                    // descriptors are always parsed insane. This will improve once we use upstream ExtParams Api.
+                    let tr = Tr::<Pk, T>::from_str_insane(s)?;
```

Comment:
In acced2756617b913046515f1fa07b53e98ea524a:

This is a bit eyebrow-raising. This means that `Descriptor::from_str` will no longer enforce any sanity rules when parsing tapscripts?

This is fine if it's temporary and we're going to fix it when we bring `ExtParams` down, but the comment should have the text `FIXME` in it so that we have a reminder to deal with it.

Reply:
Sorry, this comment was wrong/incomplete. This was meant to covenant descriptors should be parsed as `insane` because they are not safe(in the sense there is no signature guarding it).

Parsing regular descriptors should work as expected.

Reply:
Okay, perfect. Can you update the comment?